source/common/http/async_client_impl.cc

```diff
@@ -135,7 +135,10 @@ void AsyncStreamImpl::sendHeaders(HeaderMap& headers, bool end_stream) {
   is_grpc_request_ = Grpc::Common::hasGrpcContentType(headers);
   headers.setReferenceEnvoyInternalRequest(Headers::get().EnvoyInternalRequestValues.True);
   if (send_xff_) {
-    Utility::appendXff(headers, *parent_.config_.local_info_.address());
+    const Network::Address::InstanceConstSharedPtr address = parent_.config_.local_info_.address();
```

Comment:
in what case local_info_.address will be nullptr?

Reply:
FYI I already discussed this with @wgallagher offline. I think this is a mock issue in config_test that he is going to fix.